js/tidex2.js

```diff
@@ -0,0 +1,960 @@
+        //                     "open":"21449.08",
+        //                     "high":"22619.38",
+        //                     "low":"21193.59",
+        //                     "last":"21968.03",
+        //                     "vol":"188.4452385",
+        //                     "deal":"4075398.065557751",
+        //                     "change":"2"
+        //                 }
+        //             }
+        //         }
+        //     }
+        //
+        return this.parseTickers (result, symbols);
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        const response = await this.publicGetTicker (this.extend (request, params));
+        const result = this.safeValue (response, 'result');
+        //
+        //     {
+        //         "code":200,
+        //         "success":true,
+        //         "message":"",
+        //         "result":{
+        //             "name":"YFI_USDT",
+        //             "bid":"21607.842223",
+        //             "ask":"21635.50514299",
+        //             "open":"21530.31",
+        //             "high":"22619.38",
+        //             "low":"21193.59",
+        //             "last":"21644.23",
+        //             "volume":"188.4555124",
+        //             "deal":"4075612.142786252",
+        //             "change":"1"
+        //         }
+        //     }
+        //
+        return this.parseTicker (result, market);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = 1, max = 100
+        }
+        const response = await this.publicGetDepthResult (this.extend (request, params));
+        //
+        //     {
+        //         "asks":[
+        //             ["22064.13194312","0.25742"],
+        //         ],
+        //         "bids":[
+        //             ["22017.12205596","0.11"],
+        //         ]
+        //     }
+        //
+        return this.parseOrderBook (response, symbol);
+    }
+
+    parseTrade (trade, market = undefined) {
+        // fetchTrades
+        //     {
+        //         "tid":135762344,
+        //         "date":1646294384,
+        //         "price":"21991.91",
+        //         "type":"buy",
+        //         "amount":"0.0024",
+        //         "total":"52.780584"
+        //     }
+        //
+        const timestamp = this.safeTimestamp (trade, 'date');
+        const side = this.safeString (trade, 'type');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString (trade, 'amount');
```

Comment:
It seems that we can also get the trade`cost` from the response from the *total* field like so

```JavaScript
const cost = this.safeString (trade, 'total');
``` 

and then pass it into `this.safeTrade (...)` 

```JavaScript
'price': price,
'amount': amount,
'cost':  cost,
```